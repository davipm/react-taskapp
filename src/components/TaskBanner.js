import React from "react";
import PropTypes from "prop-types";

const TaskBanner = ({ userName, taskItems }) => (
  <h4 className="bg-primary text-white text-center p-4" data-testid="title">
    {userName}'s Tasks App ({taskItems.filter((task) => !task.done).length})
    tasks to do.
  </h4>
);

TaskBanner.propTypes = {
  userName: PropTypes.string.isRequired,
  taskItems: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      done: PropTypes.bool,
      id: PropTypes.number,
    }),
  ).isRequired,
};

export default TaskBanner;
