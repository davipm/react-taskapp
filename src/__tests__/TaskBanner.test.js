import React from "react";
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import TaskBanner from "../components/TaskBanner";

beforeEach(cleanup);

const name = 'Davi Pereira';
const data = [
  { name: "Task One", done: false, id: 0 },
  { name: "Task Two", done: false, id: 1 },
  { name: "Task Three", done: true, id: 2 },
  { name: "Task Four", done: false, id: 3 },
];

describe('Task Banner component', function () {
  it('should render user name and length task', function () {
    const { getByTestId } = render(<TaskBanner taskItems={data} userName={name} />);

    expect(getByTestId('title')).toBeTruthy();
    expect(getByTestId('title').textContent).toBe('Davi Pereira\'s Tasks App (3) tasks to do.');
  });

  it('should take snapshot', function () {
    const { container } = render(<TaskBanner taskItems={data} userName={name} />);
    expect(container).toMatchSnapshot();
  });
});
