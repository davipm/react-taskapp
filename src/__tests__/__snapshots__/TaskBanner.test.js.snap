// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Banner component should take snapshot 1`] = `
<div>
  <h4
    class="bg-primary text-white text-center p-4"
    data-testid="title"
  >
    Davi Pereira
    's Tasks App (
    3
    ) tasks to do.
  </h4>
</div>
`;
